code/game/machinery/genpop.dm

```diff
@@ -0,0 +1,372 @@
+//Credit to oraclestation for the idea! This just a recode...
+
+/**
+
+Genpop!
+Idea by oraclestation, re-coded by Kmc2000 with sprites by Cdey.
+
+
+*/
+
+#define MAX_TIMER 10 HOURS //Permabrig.
+#define PRESET_SHORT 2 MINUTES
+#define PRESET_MEDIUM 3 MINUTES
+#define PRESET_LONG 5 MINUTES
+
+/**
+A turnstile that allows one-way travel for people without ACCESS_PRISONER
+*/
+
+/obj/machinery/turnstile
+	name = "turnstile"
+	desc = "A mechanical door that permits one-way access to the brig."
+	icon = 'icons/obj/objects.dmi'
+	icon_state = "turnstile_map"
+	power_channel = AREA_USAGE_ENVIRON
+	density = TRUE
+	obj_integrity = 250
+	max_integrity = 250
+	//Robust! It'll be tough to break...
+	armor = list("melee" = 50, "bullet" = 20, "laser" = 0, "energy" = 80, "bomb" = 10, "bio" = 100, "rad" = 100, "fire" = 90, "acid" = 50, "stamina" = 0)
+	anchored = TRUE
+	use_power = FALSE
+	idle_power_usage = 2
+	resistance_flags = LAVA_PROOF | FIRE_PROOF | UNACIDABLE | ACID_PROOF
+	layer = OPEN_DOOR_LAYER
+	//Seccies and brig phys may always pass, either way.
+	req_one_access = list(ACCESS_BRIG, ACCESS_BRIGPHYS, ACCESS_PRISONER)
+
+//Executive officer's line variant. For rule of cool.
+/obj/machinery/turnstile/hop
+	name = "HOP line turnstile"
+	req_one_access = list(ACCESS_BRIG, ACCESS_HEADS)
+
+/obj/structure/closet/secure_closet/genpop
+	name = "genpop locker"
+	desc = "A locker to store a prisoner's valuables, that they can collect at a later date."
+	req_access = list(ACCESS_BRIG)
+	anchored = TRUE
+	var/registered_name = null
+
+/obj/structure/closet/secure_closet/genpop/Initialize(mapload)
+	. = ..()
+	//Show that it's available.
+	set_light(1,1,COLOR_GREEN)
+
+/obj/structure/closet/secure_closet/genpop/attackby(obj/item/W, mob/user, params)
+	var/obj/item/card/id/I = null
```

Comment:
```suggestion
	var/obj/item/card/id/I
```

Reply:
There's nothing wrong with assigning null imo

Reply:
It's already null there's no point